plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java

```diff
@@ -86,8 +86,9 @@ public boolean performFinish() {
 	private boolean importProject() {
 		try {
 			final DelegatingProgressMonitor delegatingMonitor = new DelegatingProgressMonitor();
-			IStatus jobResult = WizardUtils.runInWizard(
-					new ImportJob(model.getGitUrl(), model.getCloneDestination(), delegatingMonitor), delegatingMonitor, getContainer());
+			ImportJob importJob = new ImportJob(model.getGitUrl(), model.getCloneDestination(), delegatingMonitor)
+					.setGitRef(model.getGitRef());
```

Comment:
I'm not a big fan of having constructors with huge lists of parameters, the builder pattern is much better suited for that case.

I added it as a setter because, another parameter needs to be injected next (an import filter a.k.a. contextdir). Injecting that as a constructor arg, would gives us a 5 args constructor. Meh.